core/src/main/java/org/apache/iceberg/BaseTable.java

```diff
@@ -280,6 +281,11 @@ public String toString() {
     return name();
   }
 
+  @Override
+  public Optional<StatisticsFile> statistics(long snapshotId) {
+    return statisticsFiles().stream().filter(file -> file.snapshotId() == snapshotId).findFirst();
```

Comment:
See comment above on a different signature to incorporate the desired blob type a user is looking for and how I think this APi should be more best effort (return the latest statistics file that we can find) rather than returning Optional.empty